CONTRIBUTING.md

```diff
@@ -65,7 +65,37 @@ the first time it is run; however, if you are running in a [devcontainer] you ma
 azd auth login --use-device-code
 ```
 
+#### Environments
+
+You can run live tests against either Azure [Key Vault] or [Managed HSM]. A Key Vault and keys needed for testing will
+be provisioned by default considering [pricing], so if you merely run `az up` you will be prompted for a couple of
+parameters and then a Key Vault is provisioned; however, you can pre-create environments to easily switch between the two.
+
+##### Key Vault
+
+To pre-create, provision, and test an environment for Azure [Key Vault], substituting "keyvault" with any name you'd prefer:
+
+```bash
+azd env new keyvault # --location {location} --subscription {subscription}
+azd up -e keyvault
+go test ./... -args -env keyvault -live
+```
+
+##### Managed HSM
+
+To pre-create, provision, and test an environment for Azure [Managed HSM], substituting "managedhsm" with any name you'd prefer:
+
+```bash
+azd env new managedhsm # --location {location} --subscription {subscription}
+azd env set -e managedhsm AZURE_MANAGEDHSM true
```

Comment:
Oh this is clever!  make sure not set the value for the specific env. That would avoid an issue where a different env is selected in the middle of `azd env new` and `azd env set`  (b/c `azd env new` makes the new env to be selected automatically)

Reply:
I did this because I didn't realize `azd env new` sets the default. To be honest, I haven't quite figured out - not that I've put a lot of thought into it - what changes the default. I've grown distrustful and find myself always passing `-e`.